Split out JALR's unsealing behavior to Zysentry (#768)
riscv/riscv-cheri: src/cheri/riscv-unpriv-integration.adoc

Instructions that operate on capabilities will produce a result with a cleared {ctag} if the source capability is sealed and the operation would alter its address, bounds, or permissions.
Extensions that augment capability metadata must describe their interaction(s) with sealed capabilities.

Given a capability with `CT=0`,
deriving a capability with `CT≠0` is termed "sealing"
(or "sealing at ``CT``" when a particular output ``CT`` is meant).
Dually,
deriving a `CT=0` capability from a `CT≠0` capability is termed "unsealing"
(or "unsealing at ``CT``" when a particular input ``CT`` is meant).
In general, each of these actions may require *authority*
to operate at the non-zero type;
extensions will specify how software expresses this authority
for types not defined above.

[#sec_cap_type_ambient]
Ambient sealing type::
Some capability types are said to be "ambiently available" (or just "ambient")

> if they require no particular authority to use to seal a capability.

Reviewer: ```suggestion
if they don't require a specific permission to be granted to seal a capability.
```

I find "no particular authority" quite vague - I think we're saying that there's no specific sealing permission required.

Ambient sealing type::
Some capability types are said to be "ambiently available" (or just "ambient")
if they do not require specific authority to seal a capability (with that type).
For example, if <<section_rvy_sentry_insn_ext>> is available on a given platform,
the type with which it seals capabilities is considered ambiently available.

> (With the capabilities of <<app_cap_description>>, that would be type `1`.)

Reviewer 2: I would quite like the architectural otypes to be defined centrally rather than dependent on the encoding. I.e. 1 is always ambient sentry, but does not need to be encodable by CHERIoT. That way the YTYPER result is independent of the encoding.

Author: I would prefer not to have a central registry for two reasons:

- Suppose I build a machine that has University CHERI mediated seals and a plethora of types as well as the default encoding's (ambient, jalr) semantics for otype 1.  Because I built this machine in the near term, there was no meaning for otype 2, so it was perfectly reasonable to use as a software type.  In the future, suppose otype 2 is given meaning, then some later-written software might see an otype 2 capability on my machine and presume it has these new semantics.  In order to prevent that, the software has to be aware of the set of (architectural) types on my machine anyway.

- Having a central registry suggests that _extensions_ are the right place to define new types, and this means that coordination across extensions is required to avoid conflicts.  Despite that, encodings still have to pick a mapping from their bit patterns to supported architectural types.

If (architectural) otypes and their semantics instead are functions of the encoding and the ABI, it should be pretty natural to write software using symbolic names for particular otype semantics (further allowing for easy compile-time testing of the availability of particular semantics, at that) and we don't have to worry about conflicts between extensions.  But maybe I'm worried about the wrong things?

Author: Sorry, wasn't clear.  Suppose otype 2 _is_ a hardware type, but doesn't have any semantics beyond "sealed" when I build the hypothetical machine in question.
